### ISSUE TITLE
JNLP files after execution not deleted

### ISSUE THREAD

**Issue by bsachin89** [karakun/OpenWebStart#300]
Hi Team,

I observed one difference when I use JavaWebStart(JWS) and  OpenWebStart(OWS) is
in case JWS, after execution is completed, corresponding .jnlp file is deleted.
Whereas in case of OWS,  .jnlp file is not deleted.
Is it something configurable in OWS?

Thanks in advance.

**Comment by sclassen** [comment]
Hi

This is browser specific behavior.
We cannot control where the browser will save the jnlp and we do not want to delete files which have not been created by OpenWebStart.

If you are in control of the webpage which contains the link then the simplest fix is to replace the `http` protocol with `jnlp` (or jnlps if you want to ensuer SSL).
The browser will den lookup the application associated with the protocol from you OS and launch OpenWebStart if you have registered the protocol during installation of OpenWebStart (which is the default, so most likely you will be fine)

**Comment by engaric** [comment]
This is defintely an issue for more complicated applications with backend logic driving the presentation of the jnlp(x) file as updating the link to be jnlp(s) in place of http(s) may be non trivial.  Yes the jnlp file is downloaded to a location decided by the browser but it is clear it is domain of webstart to handle it.  I would also vote for this being given some attention if the idea is for OpenWebstart to be a drop in replacement for Oracle Webstart.

**Comment by Narimm** [comment]
I am also going to leave a vote in favour of this being added as a configurable option in OWS -  JWS's handling is definitively better in terms of open run delete - ideally speaking the better handling would be to force jnlp files be saved to the local tmp file directory run from there and allow the OS to handle removal.    However it seems the issue here is that when you do setup this latter method if your downloading a lot of jlnps then they dont get removed fast enough and you end up with 10

**Comment by AlBundy33** [comment]
Oracle webstart automatically deleted the jnlp-file after launch.
-> No matter where the files comes from.

This is extremly usefull if the file can be used only once (e.g. due to session-informations inside the jnlp).

It would be really nice to have a autodelete-option in OWS too.
